airflow/providers/amazon/aws/waiters/README.md

```diff
@@ -98,3 +98,5 @@ EksHook().get_waiter("all_nodegroups_deleted").wait(clusterName=cluster_name)
 
 Note that since the get_waiter is in the hook instead of on the client side, a custom waiter is
 just `hook.get_waiter` and not `hook.conn.get_waiter`.  Other than that, they should be identical.
+
+Note the custom waiter doesn't work with resource_type, only client_type is supported.
```

Comment:
Thanks for adding this.  As we discussed on Slack, we could tackle this a few ways, but none of them should be considered blocking for this PR.  As a temporary Airflow-based fix, I like the idea of adding come logic to Base AWS Hook's "get waiter" method that checks if the cal is being made against a Resource and creates a client connection to handle it if that is the case.  Longer term, it would be nice to seer why botocore doesn't support this natively and maybe submit a PR to add it... but the hook solution is a something we could do quickly in code we own.

Reply:
Yup, I'm planning to address Airflow fix in a separate PR :)

Reply:
@ferruzzi @o-nikolas @vincbeck - Please check https://github.com/apache/airflow/pull/30897